x-pack/plugins/lens/public/datasources/form_based/utils.tsx

```diff
@@ -260,18 +257,17 @@ const accuracyModeEnabledWarning = (
   ),
 });
 
-export function getShardFailuresWarningMessages(
+export function getSearchWarningMessages(
   state: FormBasedPersistedState,
   warning: SearchResponseWarning,
   request: SearchRequest,
   response: estypes.SearchResponse,
   theme: ThemeServiceStart
 ): UserMessage[] {
   if (state) {
-    if (warning.type === 'shard_failure') {
-      switch (warning.reason.type) {
-        case 'unsupported_aggregation_on_downsampled_index':
-          return Object.values(state.layers).flatMap((layer) =>
+    if (warning.type === 'incomplete') {
+      return hasUnsupportedDownsampledAggregationFailure(warning)
+        ? Object.values(state.layers).flatMap((layer) =>
```

Comment:
@dej611 can you also review/test the TSDB behavior?

Reply:
There's a specific FTR test for this case. I see it passing, so all good by me 👍